internal/restic/util_windows.go

```diff
@@ -0,0 +1,39 @@
+package restic
+
+import (
+	"github.com/restic/restic/internal/debug"
+	"golang.org/x/sys/windows"
+)
+
+// IsRunningAsAdminOnWindows is true if the process is running with admin privileges
+var IsRunningAsAdminOnWindows = func() bool {
```

Comment:
This runs unconditionally for all commands, even though only recover needs the result. As these hacks accumulate, the startup cost of all commands increases. Also, errors are logged but then ignored. It's better written as a function that reads a variable initialized Once:

```
var windowsAdmin struct {
	sync.Once
	isAdmin bool
	err     error
}

func isWindowsAdmin() (bool, error) {
	windowsAdmin.Do(func() {
		windowsAdmin.isAdmin, windowsAdmin.err = /* whatever */
	})
	return windowsAdmin.isAdmin, windowsAdmin.err
}
```

Note that the function need not be public since it is used only inside this package.

Reply:
Was already there in a uncommited version until I "simplified" it. Added it back in https://github.com/restic/restic/pull/2875/commits/a69924fb41920a49681cf4363e0a82e8a4e02ae5.